src/m365/cli/commands/cli-doctor.ts

```diff
@@ -35,13 +35,16 @@ class CliDoctorCommand extends Command {
 
   public async commandAction(logger: Logger): Promise<void> {
     const roles: string[] = [];
-    const scopes: string[] = [];
+    const scopes: Map<string, string[]> = new Map<string, string[]>();
 
     Object.keys(auth.service.accessTokens).forEach(resource => {
       const accessToken: string = auth.service.accessTokens[resource].accessToken;
 
       this.getRolesFromAccessToken(accessToken).forEach(role => roles.push(role));
-      this.getScopesFromAccessToken(accessToken).forEach(scope => scopes.push(scope));
+      const [res, scp] = this.getScopesFromAccessToken(accessToken);
```

Comment:
Hi @dojcsakj, the only thing I'm noticing here is that I'm getting multiple SharePoint entries. That's because I have a token for the root site and for the admin site. The same will fly for contoso-my.sharepoint.com

The scopes for these tokens will be the same, so seems to me a waste of lines. The list may get extremely long like this. On the other hand, it does give information that there are tokens available for these endpoints as well. 

@milanholemans, what's your take on this? Show everything? or show the scopes for the `https://contoso.sharepoint.com` audience and leave the rest?



```json
{
  "os": {
    "platform": "linux",
    "version": "#1 SMP Fri Jan 27 02:56:13 UTC 2023",
    "release": "5.15.90.1-microsoft-standard-WSL2"
  },
  "cliVersion": "7.0.0",
  "nodeVersion": "v18.17.0",
  "cliAadAppId": "31359c7f-bd7e-475c-86db-fdb8c937548e",
  "cliAadAppTenant": "common",
  "authMode": "DeviceCode",
  "cliEnvironment": "",
  "cliConfig": {
    "errorOutput": "stderr",
    "prompt": true
  },
  "roles": [],
  "scopes": {
    "https://graph.microsoft.com": [
      "AllSites.FullControl",
      "AppCatalog.ReadWrite.All",
      "AuditLog.Read.All",
      "Bookings.Read.All",
      "ChannelMember.ReadWrite.All",
      "ChannelMessage.Read.All",
      "ChannelMessage.Send",
      "ChannelSettings.ReadWrite.All",
      "Chat.Read",
      "Chat.ReadWrite",
      "Directory.AccessAsUser.All",
      "Directory.ReadWrite.All",
      "ExternalConnection.ReadWrite.All",
      "Group.ReadWrite.All",
      "IdentityProvider.ReadWrite.All",
      "Mail.Read.Shared",
      "Mail.ReadWrite",
      "Mail.Send",
      "Notes.Read.All",
      "Place.Read.All",
      "Policy.Read.All",
      "RecordsManagement.ReadWrite.All",
      "Reports.Read.All",
      "SecurityEvents.Read.All",
      "ServiceHealth.Read.All",
      "ServiceMessage.Read.All",
      "ServiceMessageViewpoint.Write",
      "Tasks.ReadWrite",
      "Team.Create",
      "TeamMember.ReadWrite.All",
      "TeamsAppInstallation.ReadWriteForUser",
      "TeamSettings.ReadWrite.All",
      "TeamsTab.ReadWrite.All",
      "TermStore.ReadWrite.All",
      "User.Invite.All",
      "User.ReadWrite.All",
      "profile",
      "openid",
      "email"
    ],
    "https://blimped.sharepoint.com": [
      "AllSites.FullControl",
      "AppCatalog.ReadWrite.All",
      "AuditLog.Read.All",
      "Bookings.Read.All",
      "ChannelMember.ReadWrite.All",
      "ChannelMessage.Read.All",
      "ChannelMessage.Send",
      "ChannelSettings.ReadWrite.All",
      "Chat.Read",
      "Chat.ReadWrite",
      "Directory.AccessAsUser.All",
      "Directory.ReadWrite.All",
      "ExternalConnection.ReadWrite.All",
      "Group.ReadWrite.All",
      "IdentityProvider.ReadWrite.All",
      "Mail.Read.Shared",
      "Mail.ReadWrite",
      "Mail.Send",
      "Notes.Read.All",
      "Place.Read.All",
      "Policy.Read.All",
      "RecordsManagement.ReadWrite.All",
      "Reports.Read.All",
      "SecurityEvents.Read.All",
      "ServiceHealth.Read.All",
      "ServiceMessage.Read.All",
      "ServiceMessageViewpoint.Write",
      "Tasks.ReadWrite",
      "Team.Create",
      "TeamMember.ReadWrite.All",
      "TeamsAppInstallation.ReadWriteForUser",
      "TeamSettings.ReadWrite.All",
      "TeamsTab.ReadWrite.All",
      "TermStore.ReadWrite.All",
      "User.Invite.All",
      "User.ReadWrite.All"
    ],
    "https://blimped-admin.sharepoint.com": [
      "AllSites.FullControl",
      "AppCatalog.ReadWrite.All",
      "AuditLog.Read.All",
      "Bookings.Read.All",
      "ChannelMember.ReadWrite.All",
      "ChannelMessage.Read.All",
      "ChannelMessage.Send",
      "ChannelSettings.ReadWrite.All",
      "Chat.Read",
      "Chat.ReadWrite",
      "Directory.AccessAsUser.All",
      "Directory.ReadWrite.All",
      "ExternalConnection.ReadWrite.All",
      "Group.ReadWrite.All",
      "IdentityProvider.ReadWrite.All",
      "Mail.Read.Shared",
      "Mail.ReadWrite",
      "Mail.Send",
      "Notes.Read.All",
      "Place.Read.All",
      "Policy.Read.All",
      "RecordsManagement.ReadWrite.All",
      "Reports.Read.All",
      "SecurityEvents.Read.All",
      "ServiceHealth.Read.All",
      "ServiceMessage.Read.All",
      "ServiceMessageViewpoint.Write",
      "Tasks.ReadWrite",
      "Team.Create",
      "TeamMember.ReadWrite.All",
      "TeamsAppInstallation.ReadWriteForUser",
      "TeamSettings.ReadWrite.All",
      "TeamsTab.ReadWrite.All",
      "TermStore.ReadWrite.All",
      "User.Invite.All",
      "User.ReadWrite.All"
    ]
  }
}
```

Reply:
My gut feeling says to show everything. However, as you mentioned, these 2 resources will always have the same scopes. So we could hide the admin resource for readability yes.

Reply:
Yeah, the idea is that you need to see the scopes. The rest is unnecessary.

@dojcsakj, could you take care that for SharePoint resources, only one is shown? It would need to be the one without -admin or -my. Maybe you'd need to format the resource url, because even if there are only admin endpoint tokens, I'd still only want to see the contoso.SharePoint.com url as resource.

Reply:
Here you are.